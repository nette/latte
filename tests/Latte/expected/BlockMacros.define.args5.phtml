<?php
%A%

final class Template%a% extends Latte\Runtime\Template
{
	protected const BLOCKS = [
		['test' => 'blockTest'],
	];


	public function main(): array
	{
		extract($this->params);
		echo 'default values

';
		if ($this->getParentName()) {
			return get_defined_vars();
		}
		echo '
a) ';
		$this->renderBlock($__nm = 'test', [1] + [], 'html');
		echo '

b) ';
		$this->renderBlock($__nm = 'test', ['var1' => 1] + [], 'html');
		return get_defined_vars();
	}


	public function blockTest(array $__args): void
	{
		extract($this->params);
		$var1 = $__args[0] ?? $__args['var1'] ??  0;
		$var2 = $__args[1] ?? $__args['var2'] ??  [1, 2, 3];
		$var3 = $__args[2] ?? $__args['var3'] ??  10;
		echo '	Variables ';
		echo LR\Filters::escapeHtmlText($var1) /* line 4 */;
		echo ', ';
		echo LR\Filters::escapeHtmlText(($this->filters->implode)($var2)) /* line 4 */;
		echo ', ';
		echo LR\Filters::escapeHtmlText($var3) /* line 4 */;
		echo "\n";
	}

}
