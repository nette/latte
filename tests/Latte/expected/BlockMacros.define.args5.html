default values


a) 	Variables 1, 123, 10


b) 	Variables 1, 123, 10
